
syntax = "proto3";

package no.mechatronics.sfi.grpc_fmu;

option java_outer_classname = "GenericFmuProto";

import "definitions.proto";

service GenericFmuService {

    rpc CreateInstance (Empty) returns (Int);

    rpc GetModelName (Empty) returns (Str);
    rpc GetModelVariables (Empty) returns (ScalarVariables);
	
    //instance methods
    rpc GetCurrentTime (Int) returns (Real);
    rpc IsTerminated (Int) returns (Bool);

    rpc Init (InitRequest) returns (Bool);
    rpc Step (StepRequest) returns (Status);
    rpc Terminate (TerminateRequest) returns (Empty);

    rpc Reset(ResetRequest) returns (Status);

    rpc Read (VarRead) returns (Var);
    rpc Write (VarWrite) returns (Status);

}
